src/api/app/models/attrib.rb

```diff
@@ -55,71 +31,54 @@ class Attrib < ActiveRecord::Base
   validates :package_id, :absence => {:message => "can't also be present"}, if: "project_id.present?"
   validates :project, presence: true, if: "package_id.nil?"
 
-  validates_with ValueLengthValidator
-  validates_with IssueValidator
-  validates_with AllowedValuesValidator
-
-  delegate :name, to: :attrib_type
-  delegate :namespace, to: :attrib_type
-
-  scope :nobinary, -> { where(:binary => nil) }
+  # Validators in app/validators
+  validates_with AttribValueLengthValidator
+  validates_with AttribIssueValidator
+  validates_with AttribAllowedValuesValidator
 
+  #### Class methods using self. (public and then private)
   def self.find_by_container_and_fullname( container, fullname )
-    atype = AttribType.find_by_name!(fullname)
-    return container.attribs.where(attrib_type: atype).first
+    container.attribs.where(attrib_type: AttribType.find_by_name!(fullname)).first
```

Comment:
I know that this was there before but why do we need the exclamation mark here? It'll raise an ActiveRecord::RecordNotFound exception if no AttribType is found ...